docx/docx.go

```diff
@@ -23,66 +24,156 @@ type OpenControlDocx struct {
 	*models.OpenControl
 }
 
-//BuildDocx exports a Doxc ssp based on a template
+//BuildDocx exports a Docx ssp based on a template
 func (config *Config) BuildDocx() error {
 	openControl := OpenControlDocx{models.LoadData(config.OpencontrolDir, "")}
 	docTemplate, err := docTemp.GetTemplate(config.TemplatePath)
 	if err != nil {
 		return err
 	}
-	funcMap := template.FuncMap{"getControl": openControl.FormatControl}
+	funcMap := template.FuncMap{
+		"getAllControlSections": openControl.FormatControl,
+		"getControlSection":     openControl.FormatControl,
+		"getParameter":          openControl.FormatParameter,
+		"getResponsibleRole":    openControl.FormatResponsibleRoles,
+	}
 	docTemplate.AddFunctions(funcMap)
 	docTemplate.Parse()
 	docTemplate.Execute(config.ExportPath, nil)
 	return err
 }
 
-// FormatControl returns a control formatted for docx
-func (openControl *OpenControlDocx) FormatControl(standardControl string) string {
+type componentInfoType int
```

Comment:
yeah. I want so create some type of enumeration of what type of data i want going into the common function `getComponentText`. i can't pass a int by itself, only through those value declared by my new type.